### PR TITLE
Make WOS deployable with flask server

### PR DESCRIPTION
**Pull request by markstur**
Adds a python flask server in a subdir.
The slack integration is started if the necessary
config exists. The web-ui is able to run and
report an error when env vars are missing.

Some refactoring to share code between the flask
server (server.py) and the command-line run.py.

Use PORT env var for Bluemix health check
Bluemix expects the server on the PORT.
So use the env var PORT when starting it.

### CODE REVIEW

File: ?
Comment (stevemar) [inline ?:1]: we probably want to add services here too, but another time

Comment (stevemar) [inline ?:1]: this whole file needs comments or a link to where it's from (flask tutorial?)

Comment (stevemar) [inline ?:1]: should this not hardcoded?

Comment (stevemar) [inline ?:1]: should reference where we're getting this from

Comment (stevemar) [inline ?:1]: should reference where we're getting this from

Comment (stevemar) [inline ?:1]: nice refactoring

Comment (stevemar) [inline ?:1]: * should reference where we're getting this from
* anyway to make this not in templates?

**Review by stevemar — COMMENTED** [review]
i misunderstood this initially, i didn't think it would deploy it's own UI. I thought it would simply run the original run.py in a more automated way.

File: ?
Comment (scottdangelo) [inline ?:1]: sender should be instantiated outside the while loop, I think.

**Review by scottdangelo — COMMENTED** [review]
Looks mostly good. See comment.

File: ?
Comment (markstur) [reply ?:1]: A few lines up the code allows channel to be set. So for now it needs to be in the loop. There probably should be some conditionals for channel and user changes (to only do stuff first time for example).

**Comment by markstur** [comment]
Took non flask/web-ui code out for deploy to bluemix.
Need to revisit comment in this and decide what to do with web-ui.

**Pull request closed without merging**